README.md

```diff
@@ -1,254 +1,422 @@
+    for each application bound to the service. This requires that the service broker is able to
+    communicate with the Conjur master for each bind request.
+
+    The advantage to this is finer-grained access control and audit logs in Conjur.
+
+3. **Permit the Application to Access Secrets in Conjur**
+
+    PCF applications can be granted access to secrets using either the Org and Space layers,
+    or with the application host identity.
+
+    #### Privilege Org and Space Layers
+
+    Applications can be granted access to secrets by privileging the Org or Space layers
+    to read secrets using Conjur policy.
+
+    The layer Ids use the Org and Space GUID identifiers, which may be obtained
+    using the Cloud Foundry CLI:
+    ```sh-session
+    $ cf org --guid <org-name>
+    6b40649e-331b-424d-afa0-6d569f016f51
+
+    $ cf space --guid <space-name>
+    72a928f6-bf7c-4732-a195-896f67bd1133
+    ```
+
+    For example, the policy to privilege a space to access a secret is:
+    ```yaml
+    - !permit
+      resource: my-secret-id
+      role: !layer cf/prod/6b40649e-331b-424d-afa0-6d569f016f51/72a928f6-bf7c-4732-a195-896f67bd1133
+      privileges: [ read, execute ]
+    ```
+
+    #### Privilege Application Host Identity
 
-### Run Your Application
+    > **NOTE:** Application Host privileging is not available when using Space Host Identies.
 
-Now that your application is privileged to access the secrets it needs in Conjur,
-start or restage the app so that the Conjur Buildpack can inject the secret values
-into the running application's environment.
+    After your application has been pushed to PCF, you can use its host identity in
+    Conjur policy to grant it access to secrets.
+
+    The host identity of the application is stored in the `authn_login` field in the `cyberark-conjur`
+    credentials in the application's environment, and might look something like
+    `host/cf/prod/0299a19d-7de4-4e98-89f6-372ac7c0521f`.
+
+      > **NOTE**: In PCF version 2.0+, the host identity includes the Organization and Space GUIDs in
+      the Host identity, for example:
+
+      ```
+      host/cf/prod/cbd7a05a-b304-42a9-8f66-6827ae6f78a1/8bf39f4a-ebde-437b-9c38-3d234b80631a/c363669e-e43b-40b9-b650-493d3bdb4663
+      ```
+
+4. **Run Your Application**
+
+    Now that your application is privileged to access the secrets it needs in Conjur,
+    start or restage the app so that the Conjur Buildpack can inject the secret values
+    into the running application's environment.
+
+    ```sh
+    cf start my-app
+
+    # or
+
+    cf restage my-app
+    ```
 
 The secrets are now available to be used by the application, but are not visible
 when you run `cf env my-app` or if you `cf ssh my-app` and run `printenv`.
 
-## Contributing
+### <a name="rotate-credentials"> Rotating Host API Keys
+
+When the API key for a PCF application host is rotated, the application needs to be re-bound
```

Comment:
Now that I think about it, for an app-level host you wouldn't even rotate the key, you would just unbind, rebind, and re-entitle. It might be good to split this section out into two subsections for the different level hosts, and relate to the env var setting (eg if you set [space level hosts to true], then ...)